resources/resource_factories/create/create.go

```diff
@@ -156,10 +161,7 @@ func (c *Client) FromRemote(uri string, options map[string]interface{}) (resourc
 
 	resourceID := helpers.HashString(uri, options)
 
-	// This caches to memory and will, in server mode, not be evicted unless the resourceID changes
-	// or the server restarts.
-	// There is ongoing work to improve this.
-	return c.rs.ResourceCache.GetOrCreate(resourceID, func() (resource.Resource, error) {
+	_, httpResponse, err := c.cacheGetResource.GetOrCreateBytes(resourceID, func() ([]byte, error) {
```

Comment:
All of this looks very goo. I one comment; I suspect it would be more (memory) effective to to use `GetOrCreate` (which returns a `io.ReadCloser` -- but then you need to remember to close it when done (or in a defer).